index.js

```diff
@@ -77,59 +77,50 @@ const styles = {
   bgWhiteBright: [107, 49]
 };
 
-const isString = val => val && typeof val === 'string';
-const unstyle = val => isString(val) ? val.replace(ansiRegex, '') : val;
-const hasOpen = (input, open) => input.slice(0, open.length) === open;
-const hasClose = (input, close) => input.slice(-close.length) === close;
-const color = (str, style, hasNewline) => {
-  const { open, close } = style;
-  if (!(hasOpen(str, open) && hasClose(str, close))) {
-    str = style.open + str.replace(style.closeRe, style.open) + style.close;
-  }
-  if (hasNewline) {
-    return str.replace(/(\r?\n)/g, `${style.close}$1${style.open}`);
-  }
-  return str;
-};
-
-function wrap(...args) {
-  if (colors.visible === false) return '';
+function wrap() {
+  if (!colors.visible) return '';
+  let out = format.apply(null, arguments);
+  if (!colors.enabled || out.trim().length === 0) return out;
 
-  let str = util.format(...args);
-  if (colors.enabled === false) return str;
-  if (str.trim() === '') return str;
+  let i=0, tmp, arr=this.stack;
+  let isMulti = !!~out.indexOf('\n');
 
-  const newline = str.includes('\n');
-  for (const key of this.stack) {
-    str = color(str, styles[key], newline);
+  for (; i < arr.length; i++) {
+    tmp = styles[arr[i]]; // { x1, x2, rgx }
+    out = tmp.open + out.replace(tmp.closeRe, tmp.open) + tmp.close;
+    isMulti && (out = out.replace(/(\r?\n)/g, `${tmp.close}$1${tmp.open}`));
   }
-  return str;
+
+  return out;
 }
 
 function style(stack) {
-  const create = (...args) => wrap.call(create, ...args);
-  create.stack = stack;
-  Reflect.setPrototypeOf(create, colors);
-  return create;
+  let ctx = {};
+  let fn = wrap.bind(ctx);
+  ctx.stack = fn.stack = stack;
+  Reflect.setPrototypeOf(fn, colors);
+  return fn;
 }
 
 function decorate(style) {
   style.open = `\u001b[${style[0]}m`;
   style.close = `\u001b[${style[1]}m`;
   style.closeRe = new RegExp(`\\u001b\\[${style[1]}m`, 'g');
-  return style;
 }
 
-for (const key of Object.keys(styles)) {
+for (let key in styles) {
   decorate(styles[key]);
   Reflect.defineProperty(colors, key, {
     get() {
-      return style(this.stack ? this.stack.concat(key) : [key]);
+      return this.stack !== void 0 ? (this.stack.push(key),this) : style([key]);
```

Comment:
Please change this to an `if` and `return` instead of the combo using the `,`.
```js
if (this.stack !== void 0) {
  this.stack.push(key);
  return this;
}
return style([key]);
```